ballerina/swiftmt_parser.bal

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import swiftmt.com.prowidesoftware.swift.io as prowide;
+
+import ballerina/data.xmldata;
+
+# Converts a SWIFT FIN message to the corresponding SWIFT MT message type or throws an error if the type is invalid.
+#
+# + finMessage - The input SWIFT FIN message string to be parsed and converted into a SWIFT MT message type.
+# + return - On success, returns one of the SWIFT MT message types (MT1XX, MT2XX, MT9XX, MTnXX).
+# In case of an error or unsupported message type, returns an error indicating the issue.
+public isolated function swiftMtParser(string finMessage) returns record {|anydata...;|}|error {
```

Comment:
Did we look into making this a dependently-typed function instead of returning `record {}`?

Reply:
previously it was a union of records

Reply:
A dependently-typed function is something like

```ballerina
public isolated function parse(string finMessage, typedesc<record {}> td) returns td|error = external;
```

This way the user wouldn't have to cast the returned value. This is assuming the user knows what it will be parsed as.

Will also have to be written as an external function.

https://ballerina.io/learn/by-example/dependent-types/

Reply:
But the main use case is where the user does not know the message type. Because what the user gets is a string and using that it is difficult to identify the message type. So that's why previously it was a union of records. Since there are 100+ message types and if we are going to provide support for them in future it will be too much of records in the union. So I decided to change it like this.